delfin/drivers/dell_emc/vnx/vnx_block/component_handler.py

```diff
@@ -443,3 +444,141 @@ def get_iscsi_ports(self):
             name = '%s-%s' % (iscsi_port.get('sp'), iscsi_port.get('port_id'))
             iscsi_port_map[name] = iscsi_port
         return iscsi_port_map
+
+    def list_masking_views(self, storage_id):
+        views = self.navi_handler.list_masking_views()
+        views_list = []
+        if views:
+            for view in views:
+                name = view.get('storage_group_name')
+                host_names = view.get('host_names')
+                lun_ids = view.get('lun_ids')
+                if name:
+                    if name == '~physical' or name == '~management':
+                        continue
+                    view_model_template = {
+                        'native_masking_view_id': view.get(
+                            'storage_group_uid'),
+                        "name": view.get('storage_group_name'),
+                        "storage_id": storage_id
+                    }
+                    if host_names and lun_ids:
+                        host_names = list(set(host_names))
+                        for host_name in host_names:
+                            host_id = host_name.replace(' ', '')
+                            for lun_id in lun_ids:
+                                view_model = copy.deepcopy(view_model_template)
+                                view_model[
+                                    'native_storage_host_id'] = host_id
+                                view_model['native_volume_id'] = lun_id
+                                view_model[
+                                    'native_masking_view_id'] = '%s_%s_%s' % (
+                                    view_model.get('native_masking_view_id'),
+                                    host_id, lun_id)
+                                views_list.append(view_model)
+                    elif lun_ids:
```

Comment:
Masking view without `native_storage_host_group_id` or `native_storage_host_id` is invalid. Please check **comment** section in function `list_masking_views() `in file `<>/delfin/drivers/driver.py`

Reply:
Modified